scripts/west_commands/runners/nrfjprog.py

```diff
@@ -73,9 +79,8 @@ def do_add_parser(cls, parser):
         parser.add_argument('--softreset', required=False,
                             action='store_true',
                             help='use reset instead of pinreset')
-        parser.add_argument('--snr', required=False,
-                            help="""Serial number of board to use.
-                            '*' matches one or more characters/digits.""")
+        parser.add_argument('--snr', required=False, dest='dev_id',
+                            help='Deprecated: use -i/--dev-id instead')
```

Comment:
What about adding a deprecation warning? This will make deprecation clearly visible allowing people to migrate to `--dev-id`. It can be achieved with something like:

```python
import argparse
import warnings


class DeprecateAction(argparse.Action):
    def __call__(self, parser, namespace, values, option_string=None):
        warnings.warn("Argument %s is deprecated" % self.option_strings)
        setattr(namespace, self.dest, values)


parser = argparse.ArgumentParser()
parser.add_argument("--dev-id", dest="dev_id")
parser.add_argument("--snr", dest="dev_id", action=DeprecateAction,
                    help="[DEPRECATED] Use --dev-id instead")

args = parser.parse_args()
print(args)
```

Reply:
Thanks @gmarull. I agree this is a welcome addition. Since I am unsure about @mbolivar-nordic's stance on this will be, I will add it as separate commit on top.